go/worker/common/p2p/peermgmt.go

```diff
@@ -162,6 +222,36 @@ func (mgr *PeerManager) watchRegistryNodes(consensus consensus.Backend) {
 			if nodeEv.IsRegistration {
 				_ = mgr.UpdateNode(nodeEv.Node)
 			}
+		case <-ticker.C:
+			mgr.logger.Debug("ticker fired")
+			func() {
+				mgr.Lock()
+				defer mgr.Unlock()
+
```

Comment:
Can stop early if no peers and then drop the check bellow.

```suggestion
if len(mgr.peers) == 0 {
  return
}
```